docs/platform/self_hosting/configuration.mdx

```diff
@@ -129,61 +233,212 @@ tolgee.smtp.username=*****************
+
+### host
+Host the Socket.IO server will listen on. (default: `0.0.0.0`)
+
+### use-redis
+Whether Tolgee should use Redis to store clients. Useful if you have multiple instances of Tolgee for load balancing.
+(default: `false`)
+
+:::info
+In a distributed environment, you **should** use a Redis server to ensure consistent broadcast of events to all
+clients. For a simple single-node deployment, this is unnecessary.
+:::
+
+:::info
+Tolgee uses [Redisson](https://github.com/redisson/redisson) to interface with the Redis server. You can find the
+properties Redisson expects [here](https://github.com/redisson/redisson/tree/56ea7f5/redisson-spring-boot-starter#2-add-settings-into-applicationsettings-file).
+:::
+
+### external-url
+Public URL where the Socket.IO server is available. Exposed to clients via the public configuration API endpoint.
+(default: `null`)
+
+### allowed-transports
+Array of allowed Socket.IO transport methods. (default: `[ "websocket", "polling" ]`). Refer to the
+[Socket.IO documentation](https://socket.io/docs/v4/how-it-works/#transports) for more information.
+
+:::tip
+To configure an array via properties, you need to use the `[index]` notation:
+```
+tolgee.socket-io.allowed-transports[0]=websocket
+tolgee.socket-io.allowed-transports[1]=polling
+```
+
+See [Spring's documentation](https://docs.spring.io/spring-boot/docs/2.6.4/reference/html/features.html#features.external-config.yaml.mapping-to-properties) for more information.
+:::
+
+## Cache
+Prefix: `tolgee.cache`
+
+At the expense of higher memory footprint, Tolgee can use a cache to reduce the stress on the database and fetch the
+data it needs faster. Cache is also used to track certain states, such as rate limits.
+
+### enabled
+Whether Tolgee should use a cache. (default: `false`)
+
+### use-redis
+Whether Tolgee should use Redis to store cache data instead of storing it in-memory. (default: `false`)
+
+:::info
+In a distributed environment, you **should** use a Redis server to ensure consistent enforcement of rate limits, as
+they heavily rely on cache. For a simple single-node deployment, in-memory cache is sufficient.
+:::
+
+:::info
+Tolgee uses [Redisson](https://github.com/redisson/redisson) to interface with the Redis server. You can find the
+properties Redisson expects [here](https://github.com/redisson/redisson/tree/56ea7f5/redisson-spring-boot-starter#2-add-settings-into-applicationsettings-file).
+:::
+
+### default-ttl
+TTL of cache data, in milliseconds. (default: `120 * 60 * 1000` = 2 hours)
+
+### caffeine-max-size
+Maximum size of the Caffeine cache. When exceeded, some entries will be purged from cache. Set to -1 to disable size
+limitation. This has no effect when Redis cache is used. See
+[Caffeine's documentation about size-based eviction](https://github.com/ben-manes/caffeine/wiki/Eviction#size-based)
+for more information. (default: `-1`)
+
+## Sentry
+Prefix: `tolgee.sentry`
```

Comment:
Not sure if we need to document sentry for self-hosted instances.

Reply:
Even for self-hosted instances, some errors are related to the environment around the server itself, so keeping track of errors can make sense there. I think documenting everything is a good practice anyways